src/stats_impl.c

```diff
@@ -474,16 +491,14 @@ stats_set(stats_handle_t *h, stats_type_t type, void *ptr) {
     char *tofree = NULL;
     if(h->str.len < len) {
       char *replace = malloc(len);
-      ck_spinlock_ticket_lock(&h->spinlock);
       tofree = h->str.value;
       h->str.value = replace;
       h->str.len = len;
-    } else {
-      ck_spinlock_ticket_lock(&h->spinlock);
     }
+    pthread_mutex_lock(&h->mutex);
```

Comment:
Crap ,this is a bug.  This lock needs to be moved to line 493 and the `else` restored.